src/index.ts

```diff
@@ -4,90 +4,109 @@
  * @copyright Yr
  * @license MIT
  */
-export type DebugFn = (str: string, ...rest: Array<any>) => void;
-
+export type DebugFn = (str: string, ...rest: any[]) => void;
+type Indexable<T = any> = { [key: string]: T };
 /**
  * Determine if 'obj1' and 'obj2' are conceptually equal optionally ignoring properties in 'ignoredObjectProps'
  */
-export function isEqual(obj1: any, obj2: any, ignoredObjectProps?: Array<string>, debug?: DebugFn): boolean {
-  ignoredObjectProps = ignoredObjectProps || [];
+export function isEqual(obj1: any, obj2: any, ignoredObjectProps: string[] = [], debug?: DebugFn): boolean {
+  if (equal(obj1, obj2)) {
+    return true;
+  }
 
-  if (equal(obj1, obj2)) return true;
+  if (isNonArrayObject(obj1) && isNonArrayObject(obj2)) {
+    return isEqualObject(obj1, obj2, ignoredObjectProps, debug);
+  }
+  return false;
+}
 
-  if (isObject(obj1) && isObject(obj2)) {
-    const keys1 = keys(obj1, ignoredObjectProps);
-    const keys2 = keys(obj2, ignoredObjectProps);
+function isEqualObject(obj1: Indexable, obj2: Indexable, ignoredObjectProps: string[], debug?: DebugFn) {
+  const ignoredProps = stringArrayToObject(ignoredObjectProps);
+  const keys1 = nonFunctionKeys(obj1);
+  const keys2 = nonFunctionKeys(obj2);
 
-    if (keys1.length != keys2.length) return false;
+  if (keys1.length != keys2.length) {
```

Comment:
This logic changes now that you no longer filter out `ignoredObjectProps`.

If `ignoredObjectProps` is `['foo', 'bar']` `obj1` has `{ name: 'Name', foo: 'foo' }` and `obj2` has `{ name: 'Name', foo: 'foo', bar: 'bar' }` they would be equal before because both would have length 1 after stripping out the ignored object properties, but since you don't strip ignored object properties any more they will now be non-equal since they have length 2 and 3 respectively.

Reply:
I guess there isn't a testcase for this case, we should add one.

Reply:
Good catch! Will add test in another PR.